### PR TITLE
feat(webhook): add parsing for x-www-form-urlencoded content type

### PR DESCRIPTION
**Pull request by Oleksiivanov**
## Description

add parsing for x-www-form-urlencoded content type;

## Related issues

https://github.com/camunda-community-hub/spring-zeebe/pull/418

closes #

https://github.com/camunda/connectors-bundle/issues/526

### CODE REVIEW

**Review by sbuettner — APPROVED** [review]

File: spring-boot-starter-camunda-connectors/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java [hunk 1]
```diff
@@ -71,7 +76,8 @@ public InboundWebhookRestController(
   public ResponseEntity<WebhookResponse> inbound(
       @PathVariable String context,
       @RequestBody(required = false) byte[] bodyAsByteArray, // raw form required to calculate HMAC
-      @RequestHeader Map<String, String> headers)
+      @RequestHeader Map<String, String> headers,
+      @RequestHeader(value = "Content-type", required = false) String contentType)
```
Comment (igpetrov) [inline spring-boot-starter-camunda-connectors/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java:80]: Let's pull it from the map `headers` from above?

```suggestion
      @RequestHeader Map<String, String> headers)
```

File: spring-boot-starter-camunda-connectors/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java [hunk 2]
```diff
@@ -87,10 +93,28 @@ public ResponseEntity<WebhookResponse> inbound(
...
-    Map bodyAsMap =
-        bodyAsByteArray == null
-            ? Collections.emptyMap()
-            : jsonMapper.readValue(bodyAsByteArray, Map.class);
+    boolean isURLFormContentType =
+        Optional.ofNullable(contentType)
+            .map(
+                contentHeaderType ->
+                    contentHeaderType.equalsIgnoreCase("application/x-www-form-urlencoded"))
+            .orElse(false);
```
Comment (igpetrov) [inline spring-boot-starter-camunda-connectors/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java:101]:
```suggestion
    boolean isURLFormContentType = 
      headers.entrySet().stream().anyMatch(e -> HttpHeaders.CONTENT_TYPE.equalsIgnoreCase(e.getKey()) && ContentType.APPLICATION_FORM_URLENCODED.getMimeType().equalsIgnoreCase(e.getValue()));
```

File: spring-boot-starter-camunda-connectors/src/test/java/io/camunda/connector/runtime/inbound/WebhookControllerPlainJavaTests.java [hunk 3]
```diff
@@ -83,7 +83,7 @@ public void multipleWebhooksOnSameContextPath() throws IOException {
     webhook.activateEndpoint(webhookB);
 
     ResponseEntity<WebhookResponse> responseEntity =
-        controller.inbound("myPath", "{}".getBytes(), new HashMap<>());
+        controller.inbound("myPath", "{}".getBytes(), new HashMap<>(), "application/json");
```
Comment (igpetrov) [inline spring-boot-starter-camunda-connectors/src/test/java/io/camunda/connector/runtime/inbound/WebhookControllerPlainJavaTests.java:86]: Let's use `ContentType.` utility class

File: spring-boot-starter-camunda-connectors/src/test/java/io/camunda/connector/runtime/inbound/WebhookControllerTestZeebeTests.java [hunk 4]
```diff
@@ -96,7 +96,7 @@ public void multipleWebhooksOnSameContextPath() throws IOException {
     webhook.activateEndpoint(webhookB);
 
     ResponseEntity<WebhookResponse> responseEntity =
-        controller.inbound("myPath", "{}".getBytes(), new HashMap<>());
+        controller.inbound("myPath", "{}".getBytes(), new HashMap<>(), "application/json");
```
Comment (igpetrov) [inline spring-boot-starter-camunda-connectors/src/test/java/io/camunda/connector/runtime/inbound/WebhookControllerTestZeebeTests.java:99]: Here as well

**Pull request merged**